src/analyticsTracking/AmplitudeProxy.js

```diff
@@ -0,0 +1,78 @@
+import {Amplitude} from '@amplitude/react-amplitude';
+import {memoize} from '@amplitude/react-amplitude/src/lib/memoize';
+import debounce from 'lodash.debounce';
+import {buttonTracking} from './eventUtils';
+import {EVENT_CONSTANTS} from '../analyticsTracking';
+
+/**
+ *
+ */
+class AmplitudeProxy extends Amplitude {
+  logToConsole = false;
+
+  constructor(props) {
+    super(props);
+    if (typeof props.debounceInterval === 'number') {
+      this.logEvent = debounce(this.dispatch, props.debounceInterval);
+    } else {
+      this.logEvent = this.dispatch;
+    }
+
+    this._renderPropParams = {
```

Comment:
yes but that seemed like a more disparate approach, whereas with this approach they are all called through a central source and allow for the interchange of a `logger` in case we wanted to include more than one tracking service (i.e. GoogleAnalytics, HotJars). This also ensures that all events are processed through the custom `dispatch` method